libs/Microsoft.MixedReality.WebRTC/AudioTrackSource.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.MixedReality.WebRTC.Interop;
+
+namespace Microsoft.MixedReality.WebRTC
+{
+    /// <summary>
+    /// Configuration to initialize capture on a local audio device (microphone).
+    /// </summary>
+    public class LocalAudioDeviceInitConfig
+    {
+        /// <summary>
+        /// Enable automated gain control (AGC) on the audio device capture pipeline.
+        /// </summary>
+        public bool? AutoGainControl = null;
+    }
+
+    /// <summary>
+    /// Audio source for WebRTC audio tracks.
+    /// 
+    /// The audio source is not bound to any peer connection, and can therefore be shared by multiple audio
+    /// tracks from different peer connections. This is especially useful to share local audio capture devices
+    /// (microphones) amongst multiple peer connections when building a multi-peer experience with a mesh topology
+    /// (one connection per pair of peers).
+    /// 
+    /// The user owns the audio track source, and is in charge of keeping it alive until after all tracks using it
+    /// are destroyed, and then dispose of it. The behavior of disposing of the track source while a track is still
+    /// using it is undefined. The <see cref="Tracks"/> property contains the list of tracks currently using the
+    /// source.
+    /// </summary>
+    /// <seealso cref="LocalAudioTrack"/>
+    public class AudioTrackSource : IDisposable
+    {
+        /// <summary>
+        /// A name for the audio track source, used for logging and debugging.
+        /// </summary>
+        public string Name
+        {
+            get
+            {
+                // Note: the name cannot change internally, so no need to query the native layer.
+                // This avoids a round-trip to native and some string encoding conversion.
+                return _name;
+            }
+            set
+            {
+                AudioTrackSourceInterop.AudioTrackSource_SetName(_nativeHandle, value);
+                _name = value;
+            }
+        }
+
+        /// <summary>
+        /// List of local audio tracks this source is providing raw audio frames to.
+        /// </summary>
+        public List<LocalAudioTrack> Tracks { get; private set; } = new List<LocalAudioTrack>();
```

Comment:
This should be an `IEnumerable` property (or users might add stuff to the list) and possibly contain an instance of some read-only enumerable internal class (or users might cast the property value to a List and still wreak havoc - a bit defensive but the only way to be sure).

Reply:
Seems `IReadOnlyList<T>` is even better as it exposes `[]` and `Count`, which are always handy to have for the user.

Reply:
Fixed.